### PR TITLE
fix(cache): catch redis set value errors

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Catch thrown errors when trying to set a cache value when using redis

## Context
Prevent aborting run due to failure while setting value in Redis
Example
```
{
	"name": "renovate",
	"hostname": "c0cc38dbabd4",
	"pid": 9,
	"level": 30,
	"logContext": "c07cb5de9f02460c86ed63a00bcc8f85",
	"repository": "...",
	"msg": "External host error causing abort - skipping",
	"time": "2023-03-30T11:25:44.526Z",
	"v": 0
},
{
	"name": "renovate",
	"hostname": "c0cc38dbabd4",
	"pid": 9,
	"level": 20,
	"logContext": "c07cb5de9f02460c86ed63a00bcc8f85",
	"datasource": "docker",
	"packageName": "mcr.microsoft.com/mssql/server",
	"err": {
		"message": "The client is closed",
		"stack": "Error: The client is closed
    at Commander._RedisClient_sendCommand (/opt/buildpack/tools/renovate/35.23.3/node_modules/@redis/client/dist/lib/client/index.js:482:31)
    at Commander.commandsExecutor (/opt/buildpack/tools/renovate/35.23.3/node_modules/@redis/client/dist/lib/client/index.js:193:154)
    at Commander.BaseClass.<computed> [as set] (/opt/buildpack/tools/renovate/35.23.3/node_modules/@redis/client/dist/lib/commander.js:8:29)
    at Object.set (/opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/util/cache/package/redis.ts:68:17)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.set (/opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/util/cache/package/index.ts:49:3)
    at /opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/util/cache/package/decorator.ts:123:7
    at DockerDatasource.getReleases (/opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/modules/datasource/docker/index.ts:1156:20)
    at getRegistryReleases (/opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/modules/datasource/index.ts:73:15)
    at fetchReleases (/opt/buildpack/tools/renovate/35.23.3/node_modules/renovate/lib/modules/datasource/index.ts:299:15)"
	},
	"msg": "Datasource unknown error",
	"time": "2023-03-30T11:25:51.653Z",
	"v": 0
}
```

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by zharinov — DISMISSED** [review]

File: lib/util/cache/package/redis.ts [hunk 1]
```diff
@@ -65,15 +65,19 @@ export async function set(
...
+      getKey(namespace, key),
+      JSON.stringify({
+        compress: true,
+        value: await compress(JSON.stringify(value)),
+        expiry: DateTime.local().plus({ minutes: ttlMinutes }),
+      }),
+      { EX: redisTTL }
+    );
+  } catch (err) {
+    logger.debug({ err, namespace, key }, 'Error setting cache value');
```
Comment (rarkins) [inline lib/util/cache/package/redis.ts:79]: This could really blow out the logs of the Redis cache is erroring. Maybe we can reduce it?

Comment (zharinov) [reply lib/util/cache/package/redis.ts:79]: `logger.once.debug`

Comment (rarkins) [reply lib/util/cache/package/redis.ts]: Will that dedupe bass on the message or the metadata too?

Comment (zharinov) [reply lib/util/cache/package/redis.ts]: Second time this line is hit, it won't print anything even with new metadata.

**Review by rarkins — APPROVED** [review]

**Pull request merged**